lib/shopify_api/session.rb

```diff
@@ -67,10 +67,10 @@ def encoded_params_for_signature(params)
       end
     end
 
-    def initialize(url, token = nil, extra = {})
+    def initialize(url, token = nil, extra = nil)
       self.url = self.class.prepare_url(url)
       self.token = token
-      self.extra = extra
+      self.extra = extra || {}
```

Comment:
It would not be necessary check if `extra` was set if you let the `initialize` method the way it was.